src/gha_runner/clouddeployment.py

```diff
@@ -122,8 +118,8 @@ def _build_aws_params(self, count: int, user_data_params: dict) -> dict:
         params = {
             "ImageId": self.image_id,
             "InstanceType": self.instance_type,
-            "MinCount": count,
-            "MaxCount": count,
+            "MinCount": 1,
+            "MaxCount": 1,
```

Comment:
This is largely because `boto3` is not doing much shaping of the JSON objects before they are sent. These are captured directly in the API docs here: https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_RunInstances.html

Reply:
As for the defaults, it is noted in [these docs[(https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/ec2/client/run_instances.html) that the min and max are required parameters so we set them intentionally. I believe the docs you sent are for an older version of `boto`, not `boto3`.

Reply:
good catch; my googling failed me there!